Self-Consistent-Field/RHF_Gradient.py

```diff
@@ -0,0 +1,176 @@
+"""
+This script calculates nuclear gradients of RHF using
+derivative integrals obtained from PSI4
+
+"""
+
+__authors__ = "Ashutosh Kumar"
+__credits__ = ["Ashutosh Kumar"]
+
+__copyright__ = "(c) 2014-2017, The Psi4NumPy Developers"
+__license__ = "BSD-3-Clause"
+__date__ = "2017-12-17"
+
+import time
+import numpy as np
+np.set_printoptions(precision=15, linewidth=200, suppress=True)
+import psi4
+
+def rhf_gradient(mol, options):
+
+    psi4.core.set_active_molecule(mol)
+    psi4.set_options(options)
+
+    rhf_e, wfn = psi4.energy('SCF', return_wfn=True)
+
+    # Assuming C1 symmetry    
+    occ = wfn.doccpi()[0]
+    nmo = wfn.nmo()
+
+    C = wfn.Ca()
+    npC = np.asarray(C)
+
+    mints = psi4.core.MintsHelper(wfn.basisset())
+    H_ao = np.asarray(mints.ao_kinetic()) + np.asarray(mints.ao_potential())
+
+    # Update H, transform to MO basis 
+    H = np.einsum('uj,vi,uv', npC, npC, H_ao)
+
+    # Integral generation from Psi4's MintsHelper
+    MO = np.asarray(mints.mo_eri(C, C, C, C))
+    # Physicist notation    
+    MO = MO.swapaxes(1,2)
+
+    F = H + 2.0 * np.einsum('pmqm->pq', MO[:, :occ, :, :occ])
+    F -= np.einsum('pmmq->pq', MO[:, :occ, :occ, :])
+    natoms = mol.natom()
+    cart = {}; cart[0] = '_X'; cart[1] = '_Y'; cart[2] = '_Z'
+    oei_dict = {"S" : "OVERLAP", "T" : "KINETIC", "V" : "POTENTIAL"}
+
+    deriv1_mat = {}
+    deriv1_np = {}
+
+    Gradient = {};
+
+    Gradient["N"] = np.zeros((natoms, 3))
+    Gradient["S"] = np.zeros((natoms, 3))
+    Gradient["S'"] = np.zeros((natoms, 3))
+    Gradient["V"] = np.zeros((natoms, 3))
+    Gradient["T"] = np.zeros((natoms, 3))
+    Gradient["J"] = np.zeros((natoms, 3))
+    Gradient["K"] = np.zeros((natoms, 3))
+    Gradient["Total"] = np.zeros((natoms, 3))
+
+    Gradient["N"] = np.asarray(mol.nuclear_repulsion_energy_deriv1([0,0,0]))
+
+    psi4.core.print_out("\n\n")
+    Mat = psi4.core.Matrix.from_array(Gradient["N"])
+    Mat.name = "NUCLEAR GRADIENT"
+    Mat.print_out()
+    
+
+    # 1st Derivative of OEIs 
+
+    for atom in range(natoms):
+        for key in  oei_dict:
+            deriv1_mat[key + str(atom)] = mints.mo_oei_deriv1(oei_dict[key], atom, C, C)
+            for p in range(3):
+                map_key = key + str(atom) + cart[p]
+                deriv1_np[map_key] = np.asarray(deriv1_mat[key + str(atom)][p])
+                if key == "S":
+                    Gradient[key][atom][p] = -2.0 * np.einsum("ii,ii->", F[:occ,:occ], deriv1_np[map_key][:occ,:occ])
```

Comment:
Since these are arrays can you index them as `Gradient[key][atom, p]`? This goes for all arrays.